system/default_ad_api_helpers/ad_api_adaptors/src/routing_adaptor.cpp

```diff
@@ -50,6 +44,39 @@ RoutingAdaptor::RoutingAdaptor(const rclcpp::NodeOptions & options)
 
 void RoutingAdaptor::on_timer()
 {
+  const rclcpp::Time current_time = this->get_clock()->now();
+  auto fixed_goal_msg = sub_fixed_goal_.takeData();
+  auto rough_goal_msg = sub_rough_goal_.takeData();
+  auto waypoint_msg = sub_waypoint_.takeData();
```

Comment:
This change reduces waypoint throughput. Only one waypoint can be added per timer period. Also, information about the order of arrival of waypoints and goals is lost. Waypoints and goal must be processed in the order in which they arrive.

Reply:
Thank you for taking the time to review my PR.
I understand your concerns regarding the usage of take(). 
Therefore, I will close this PR.